semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -460,6 +489,24 @@ private static Optional<String> getHttpSchemeFromRawAttributes(
         return optionalExtractedProtoValue;
       }
     }
+
+    AttributeValue httpRequestOriginValue = attributeValueMap.get(HTTP_REQUEST_ORIGIN);
+    if (httpRequestOriginValue != null && !StringUtils.isEmpty(httpRequestOriginValue.getValue())) {
+      String origin = httpRequestOriginValue.getValue();
+      /** Syntax: Origin: null Origin: <scheme>://<hostname> Origin: <scheme>://<hostname>:<port> */
+      if (StringUtils.isNotEmpty(origin)) {
+        try {
+          String scheme = new URI(origin).getScheme();
+          return Optional.of(scheme);
+        } catch (URISyntaxException e) {
+          LOGGER.warn(
+              "On extracting scheme, received an invalid origin header: {}, {}",
+              origin,
+              e.getMessage());
+        }
+      }
+    }
+
```

Comment:
Technically the below for loop on SCHEME_ATTRIBUTES could return the HTTP_REQUEST_ORIGIN attribute which is malformed and not a scheme

Reply:
removed from scheme attributes